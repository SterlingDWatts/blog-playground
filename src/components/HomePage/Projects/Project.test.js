import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Project from "./Project";

describe(`Project Component`, () => {
  const project = (
    <Project
      name="Quaff Quizz"
      pic="https://sterlingdwatts.github.io/portfolio/images/quaff-quiz-mobile.png"
      site="https://sterlingdwatts.github.io/quaff_quizz"
      repo="https://github.com/SterlingDWatts/quaff_quizz"
      blurb="A fun wine quiz web app with over 20 questions ranging over 4 topics."
      moreInfo="Each time you take the quiz, you will see a different combination of answers in a different order. If you need to focus on a particular topic, you can choose from popular subjects instead!"
      tech="Tech: HTML5, CSS3, javaScript, jQuery"
    />
  );
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(project, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const tree = renderer.create(project).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
