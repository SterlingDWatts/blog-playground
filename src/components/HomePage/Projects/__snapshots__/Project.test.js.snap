// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Component renders the UI as expected 1`] = `
<div
  className="Project"
  id="projects"
>
  <header
    className="Project__header"
  >
    <h3>
      Quaff Quizz
    </h3>
  </header>
  <div
    className="Project__pic Quaff Quizz_pic"
  >
    <div
      className="Project__about"
    >
      <h5
        className="Project__name_and_links"
      >
        Quaff Quizz
         | 
        <a
          href="https://sterlingdwatts.github.io/quaff_quizz"
          rel="noopener noreferrer"
          target="_blank"
        >
          Live App
        </a>
         | 
        <a
          href="https://github.com/SterlingDWatts/quaff_quizz"
          rel="noopener noreferrer"
          target="_blank"
        >
          Repo
        </a>
      </h5>
      <p>
        A fun wine quiz web app with over 20 questions ranging over 4 topics.
      </p>
      <button
        className="Project__more_info"
        onClick={[Function]}
      >
        more info
      </button>
    </div>
  </div>
</div>
`;
